### PR TITLE
Issue 4692 Plus Updates

### PR DESCRIPTION
**Pull request by kingthorin**
Address specific false negative, plus merge all interim updates from Reflected XSS scanner.

TestPersistentXSSAttack > Updated with new/missing functionality.
ascanrules.html > Updated to align with current functionality.
Messages.properties > Added necessary keys/values to support updated functionality.
ZapAddOn.xml > Updated changes section.

### CODE REVIEW

File: src/org/zaproxy/zap/extension/ascanrules/TestPersistentXSSAttack.java [hunk 1]
```diff
@@ -126,16 +145,22 @@ public void init() {
 			return null;
 		}
 
-		HtmlContextAnalyser hca = new HtmlContextAnalyser(sinkMsg2);
+		HtmlContextAnalyser hca = new HtmlContextAnalyser(sourceMsg2);
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/ascanrules/TestPersistentXSSAttack.java]: Was this change on purpose?

File: src/org/zaproxy/zap/extension/ascanrules/resources/Messages.properties [hunk 2]
```diff
@@ -29,6 +29,9 @@ ascanrules.testscrosssitescriptv2.otherinfo.nothtml = Raised with LOW confidence
 
 ascanrules.testpersistentxssattack.name=Cross Site Scripting (Persistent)
 ascanrules.testpersistentxssattack.otherinfo=Source URL: {0}
+ascanrules.testpersistentxssattack.json.name = Cross Site Scripting Weakness (Reflected in JSON Response)
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/ascanrules/resources/Messages.properties]: This might need a tweak (same for description), `Persistent in ...`?

Comment (kingthorin) [reply src/org/zaproxy/zap/extension/ascanrules/resources/Messages.properties]: Thankd

File: src/org/zaproxy/zap/extension/ascanrules/TestPersistentXSSAttack.java (hunk 1 above)
Comment (kingthorin) [reply src/org/zaproxy/zap/extension/ascanrules/TestPersistentXSSAttack.java]: Probably overzealous copy/pasting. Will review.

**Comment by kingthorin** [comment]
Tweaked

**Comment by thc202** [comment]
Thank you!

**Comment by kingthorin** [comment]
@psiinon @denniskniep look good?
